pkg/api/controllers/requests/request.go

```diff
@@ -0,0 +1,47 @@
+package requests
+
+import (
+	"chuck-jokes/pkg/validator"
+	"github.com/gin-gonic/gin"
+	"log"
+	"strconv"
+)
+
+type IRequest interface {
+	NewCreateCategory(c *gin.Context) (*CreateCategory, []error)
+	NewLogin(c *gin.Context) (*LoginRequest, []error)
+	NewAddFavouriteRequest(c *gin.Context) (*AddFavourite, []error)
+	NewFavourites(c *gin.Context) (*Favourites, error)
+	NewPagination(c *gin.Context) PaginationRequest
+	NewAddToCategory(c *gin.Context) (*AddToCategory, []error)
+	NewJokeOfADay(c *gin.Context) (*JokeOfADay, []error)
+	NewRegister(c *gin.Context) (*Register, []error)
+	NewSetAccess(c *gin.Context) (*SetAccess, []error)
+	NewGetCategory(c *gin.Context) (*GetCategory, []error)
+}
+
+type Request struct {
```

Comment:
Czy to jest Request czy validator, ten request nie nosi za soba informacji zadnej, wiec imo to powinien byc validator lub requestValidator. Jedyne co trzyma w sobie to informacje na temat structa validator (co powinno byc interfejsem moim zdaniem, gdyz tutaj ograniczyles mozliwosci implementacji validatora, mozemy przekazac tylko ten jeden z tego structa, a jak request z innego zrodla wymagalby innej walidacji to ucina nam to mozliwosc przekazania go tutaj). Mam na mysli to ze NewRequest powinno przyjmowac interface Validator implementujacy metode Validate().